src/elements/flag/flag.js

```diff
@@ -0,0 +1,19 @@
+(function () {
+    'use strict';
+
+    angular
+        .module('semantic.ui.elements.flag', [])
+        .directive('smFlag', smFlag);
+
+    function smFlag($compile) {
+        return {
+            restrict: 'E',
+            replace: true,
+            scope: { value: '='},
+            link: function (scope, element) {
+                element.html('').append($compile('<i class="flag ' + scope.value + '"></i>')(scope));
```

Comment:
other notes: `replace` does nothing if no template is defined, which I guess is why you've got the `element.html('')` there